sway/input/keyboard.c

```diff
@@ -387,7 +387,7 @@ void sway_keyboard_configure(struct sway_keyboard *keyboard) {
 	wlr_keyboard_set_keymap(wlr_device->keyboard, keyboard->keymap);
 
 	xkb_mod_mask_t locked_mods = 0;
-	if (!input_config || input_config->xkb_numlock != 0) {
+	if (input_config && input_config->xkb_numlock) {
```

Comment:
This should test for ```xkb_numlock > 0```. The variable has three states:
-1: undefined, keep xkbcommon default
0 numlock disabled
1 numlock enabled